Update README.md - theme MIMIC

Danieleeffe1 opened the pull request:
[theme-mimic.zip](https://github.com/OpenMage/OpenMage-Themes/files/6440725/theme-mimic.zip)
RWD_LUMA updated package (v1.1)

I have created an updated and complete package for the project which I have decided to call MIMIC.

MIMIC is a theme based on RWD_default and is inspired by the LUMA theme of Magento2

MIMIC doesn't just imitate LUMA

MIMIC is Optimization for mobile devices and more in detail;

- Menu usability in mobile formats has been improved, making the menu always visible.

- A scrolltop beta module downloaded for free from magentostore for magento 1 has been added when it was still active.

The full version of the module in question with advanced features can be purchased at this address:
https://store.neklo.com/scroll-up-extension-for-magento.html

Despite being a Beta the module works well

This version of the scrolltop has a custom style to better suit the MIMIC theme.

- In the RWD theme, the field where the quantities of the products to be purchased are entered are of the "InputText" type in MIMIC they have been changed to "NUMERIC" to facilitate use with the numeric keypad of smartphones and tablets.

MIMIC does not add jQuery code other than RWD therefore it should not cause conflicts with modules already tested with RWD

INSTALLATION:
Download the package from github

If using filezilla:
Unzip the package in a directory of your choice
Open zilla files and drag and overwrite the APP and SKIN folders to the OpenMage root

If you upload the package directly to the server be sure to unzip the package to a temporary folder for safety.

Copy and overwrite the APP and SKIN folders in the OpenMage root.

SETUP
From the Openmage Admin

Go to: SYSTEM> CONFIGURATION> GENERAL> DESIGN
Leave unchanged
   Package / package name
   Themes / translations
   Themes / default

Change:
   Themes / Templates from "default" to "mimic"
   Themes / Skins (images-css) from "default" to "mimic"
   Themes / Layout from "default" to "mimic"
  
Go to: SYSTEM> CONFIGURATION> NEKLO> scroll to top to enable the module.

Flyingmana commented:
checking and adding the zip file you attached to the repository takes a lot of time, you have to be patient till a volunteer takes this task over. to create a proper PullRquest out of it

Danieleeffe1 commented:
I thank you for the answer.  the theme is based on Rwd.  if you wish I can
provide only the css and .phtml files with the modified lines indicated.

Il gio 10 giu 2021, 09:21 Daniel Fahlke ***@***.***> ha
scritto:

> checking and adding the zip file you attached to the repository takes a
> lot of time, you have to be patient till a volunteer takes this task over.
> to create a proper PullRquest out of it
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenMage/OpenMage-Themes/pull/1#issuecomment-858379422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALCKSZTQRJBGCZDC5POWDHDTSBRXDANCNFSM44JOGEZQ>
> .
>

kiatng commented:
@Danieleeffe1 Thanks for making MIMIC available to the community. I downloaded your hefty zip and took a peek.  Would you consider making MIMIC fallback to `rwd`? There are many advantages, one of which is the need to include every file from `rwd`. To allow theme fallback, replace `app\design\frontend\rwd\mimic\etc\theme.xml` with this

```xml
<!-- app\design\frontend\rwd\mimic\etc\theme.xml -->
<theme>
    <parent>rwd/default</parent>
</theme>
```
After which, you only need to retain the files which are different from `rwd` in the design and skin folders. This will make maintenance a lot easier should someone decide to use MIMIC. Also, from security stand point, it reduces the attack surface.

Danieleeffe1 commented:
kiatng

Thanks for the reply.

I had decided to propose an independent and complete file to give the possibility to those who want to modify it at their own pleasure without being bound to RWD

I have never done the procedure you suggest, I can try to do it.

I can clean up of mimic and publish only the added and changed files.

as soon as available I will post it at the end of this chat.

Danieleeffe1 commented:
"MIMIC" update

This file contains only and exclusively modified MIMIC files which differ from RWD to facilitate the control work.

I edited the theme.xml file as indicated by kiatng.

I have not changed the contents of the .xml files in Layout referring to the folders contained in the theme as I do not know the syntax needed to clean these files.

I have updated the header.phtml file to a new version

in this new version in mobile mode I managed to hide the logo previously above the menu-bar to save space and I replaced the word MENU with the "alt image" of the logo in order to get the wording you want (I would have put the store name but I could not find the correct php node).

I also optimized the css based.
[theme-mimic_minimal.zip](https://github.com/OpenMage/OpenMage-Themes/files/6631301/theme-mimic_minimal.zip)

kiatng commented:
@Danieleeffe1 Thanks so much for the minimized version, it is much easier to review. I'm interested to test it but my workload is exremely high at the moment for the next couple of months. 

Another suggestion: there are 2 modules in `app\code\community\Neklo`, it'll help interested reviewers if you would provide descriptions on what those modules are doing. In particular the `core` module, at first glance, it seems to be a little intrusive.

Danieleeffe1 commented:
> @ Danielleeffe1 Grazie mille per la versione ridotta a icona, è molto più facile da rivedere. Sono interessato a testarlo, ma il mio carico di lavoro è estremamente alto al momento per i prossimi due mesi.
> 
> Un altro suggerimento: ci sono 2 moduli in `app\code\community\Neklo`, sarà utile ai revisori interessati se fornireste descrizioni su cosa stanno facendo quei moduli. In particolare il `core`modulo, a prima vista, sembra essere un po' invadente.

kind kiatng

the module was present in free license in magentostore before it was closed.

I quote original description of mimic:

"A scrolltop beta module downloaded free from magentostore for magento 1 was added when the store was still active.

The full version of the module in question with advanced features can be purchased at this address:

[https://store.neklo.com/scroll-up-extension-for-magento.html ](https://store.neklo.com/scroll-up-extension-for-magento.html )"

I don't know the way it works, I'm not a developer.
I found a free and very simple scroll top for magento 2.

[https://store.neklo.com/scroll-up-extension-for-magento.html ](https://magefan.com/blog/magento-2-scroll-to-top-free-solution
)
This scroll top can be created as a static block in a simple widget.
I have tried to migrate this script to OpenMage but it doesn't work.

Danieleeffe1 commented:
add media.phtml in app/design/rwd/mimic/template/catalog/product/view 

in this file add two function :
1) label (for percentual sales)
2) add viedo file under image gallery (is necessary create attribute txt for insert url)

`<?php
/**
 * Magento
 *
 * NOTICE OF LICENSE
 *
 * This source file is subject to the Academic Free License (AFL 3.0)
 * that is bundled with this package in the file LICENSE_AFL.txt.
 * It is also available through the world-wide-web at this URL:
 * http://opensource.org/licenses/afl-3.0.php
 * If you did not receive a copy of the license and are unable to
 * obtain it through the world-wide-web, please send an email
 * to license@magento.com so we can send you a copy immediately.
 *
 * DISCLAIMER
 *
 * Do not edit or add to this file if you wish to upgrade Magento to newer
 * versions in the future. If you wish to customize Magento for your
 * needs please refer to http://www.magento.com for more information.
 *
 * @category    design
 * @package     rwd_default
 * @copyright   Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
 * @license     http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
 */

/**
 * Product media data template
 *
 * @see Mage_Catalog_Block_Product_View_Media
 */

/* @var $this Mage_Catalog_Block_Product_View_Media */
?>
<?php
    $_product = $this->getProduct();
    $_helper = $this->helper('catalog/output');
?>
<div class="product-image product-image-zoom">
    <div class="product-image-gallery">
        <img id="image-main"
             class="gallery-image visible"
             src="<?php echo $this->helper('catalog/image')->init($_product, 'image') ?>"
             alt="<?php echo $this->escapeHtml($this->getImageLabel()) ?>"
             title="<?php echo $this->escapeHtml($this->getImageLabel()); ?>" />
 <?php
 
$_product = Mage::getModel('catalog/product')->load($_product->getId());
 
$attribute = $_product->getResource()->getAttribute('label');
 
 if(is_object($attribute)) {
 
  $attdefaultvalue = $attribute->getDefaultValue();
 
  $attributeValue = Mage::getModel('catalog/product')->load($_product->getId())->getlabel();
 
  $attributeValueName = Mage::getModel('catalog/product')->load($_product->getId())->getAttributeText('label');
 
   if(strlen($attributeValue) &&  ($attdefaultvalue !== $attributeValue)) { 
 
    $labeltype = $attributeValueName;        
 
    } 
 
    else {
 
    $labeltype = "";
 
   } 
 
 } 
 
?>
 
<?php if ($labeltype): ?>
 
 <div id="label-attribute"><span class="<?php echo $labeltype; ?>-label"><?php echo $labeltype; ?></span></div><!--label-attribute-->  
 
<?php endif; ?>
<?php if($_product->getVideo()): ?>
        <iframe width="100%" height="315"
           src="<?php echo $_product->getVideo() ?>"
 frameborder="0" allowfullscreen></iframe>
 <?php endif; ?>
        <?php $i=0; foreach ($this->getGalleryImages() as $_image): ?>
            <?php if ($this->isGalleryImageVisible($_image)): ?>
            <img id="image-<?php echo $i; ?>"
                 class="gallery-image"
                 src="<?php echo $this->getGalleryImageUrl($_image); ?>"
                 data-zoom-image="<?php echo  $this->getGalleryImageUrl($_image); ?>" />
            <?php endif; ?>
        <?php $i++; endforeach; ?>
         
    </div>
</div>

<?php if (count($this->getGalleryImages()) > 0): ?>
<div class="more-views">
    <h2><?php echo $this->__('More Views') ?></h2>
    <ul class="product-image-thumbs">
    <?php $i=0; foreach ($this->getGalleryImages() as $_image): ?>
        <?php if ($this->isGalleryImageVisible($_image)): ?>
        <li>
            <a class="thumb-link" href="#" title="<?php echo $this->escapeHtml($_image->getLabel()) ?>" data-image-index="<?php echo $i; ?>">
                <img src="<?php echo $this->helper('catalog/image')->init($this->getProduct(), 'thumbnail', $_image->getFile())->resize(75); ?>"
                     width="75" height="75" alt="<?php echo $this->escapeHtml($_image->getLabel()) ?>" />
            </a>
        </li>
        <?php endif; ?>
    <?php $i++; endforeach; ?>
    </ul>
</div>
<?php endif; ?>

<?php echo $this->getChildHtml('after'); ?>
`

Danieleeffe1 commented:
New Update:

i have modifed the css file, now i n the page product the button add-to-link wishlist and the button add-to-link compare in the product page have the icon heart and icon grafics.

is sufficents add the follow files modifed:

style.ccs in the pat skin/frontend//rwd/mimic/css/
xicon_sprites.png in the pat skin/frontend/rwd/mimic/images/

[style-and-xicon-sprite.zip](https://github.com/OpenMage/OpenMage-Themes/files/6892258/style-and-xicon-sprite.zip)

Danieleeffe1 commented:
Latest "MINI" version with all previous improvements.

Compatible with OpenMage onwards to be installed nested in the RWD folder (I deleted the Scrolltop module, for those who want to download the previous version of mimic):

[theme-mimic_minimal.zip](https://github.com/OpenMage/OpenMage-Themes/files/7114678/theme-mimic_minimal.zip)

Danieleeffe1 commented:
> @Danieleeffe1 Thanks so much for the minimized version, it is much easier to review. I'm interested to test it but my workload is exremely high at the moment for the next couple of months.
> 
> Another suggestion: there are 2 modules in `app\code\community\Neklo`, it'll help interested reviewers if you would provide descriptions on what those modules are doing. In particular the `core` module, at first glance, it seems to be a little intrusive.

Hello Kiatng

I have loaded the latest essential version of Mimic.

I deleted the Neklo module which inserts the TopScroll function (to make your life easier) and deleted the file /app/design/frontend/rwd/mimic/layout/page.xml which prevented the loading of css files with OpenMage 20.0.11 and above .

If you read the various updates of the post history you will see that I have added a couple of changes.
